tests/figures/chart/chart_plugin.test.ts

```diff
@@ -2829,15 +2856,16 @@ describe("trending line", () => {
     const runtime = model.getters.getChartRuntime("1") as LineChartRuntime;
     const step = (6 - 1) / 25;
     //@ts-ignore
-    const data = runtime.dataSetsValues[1].data;
-    for (let i = 0; i < data.lenght; i++) {
-      const value = data.lenght;
+    const data = runtime.chartJsConfig.data.datasets[1].data;
+    for (let i = 0; i < data.length; i++) {
```

Comment:
should be a dedicated fix, this is unrelated to the commit goal:
- the ts-ignore  (and those with the following tests) should be removed  (you'll need to type cast every runtime but at least it will have some typing)
- AFAICS this stems from the weird return value of `createLineOrScatterChartRuntime` which returns more than what we actually need to create a chart runtime. I think it's worth changing its return type as well, we just create an object with unused keys?

Reply:
But I had to fix them here as the test doesn't pass anymore with some changes I made in this commit :/ Anyway, I should make another PR to fix it before

Reply:
Still in the same commit?